Python/suggestions.c

```diff
@@ -181,9 +181,9 @@ offer_suggestions_for_name_error(PyNameErrorObject *exc) {
 PyObject *_Py_Offer_Suggestions(PyObject *exception) {
     PyObject *result = NULL;
     assert(!PyErr_Occurred());
-    if (PyErr_GivenExceptionMatches(exception, PyExc_AttributeError)) {
+    if (Py_TYPE(exception) == (PyTypeObject*)PyExc_AttributeError) {
```

Comment:
You can use Py_IS_TYPE(). Or you can get the type into a variable (exc_type?) and then compare it. Or you can leave it as it is :-)